examples/Raytracer_Benchmark/Raytracer.cpp

```diff
@@ -137,28 +139,26 @@ adept::Color_t RaytraceOne(RaytracerData_t const &rtdata, Ray_t &ray, int genera
 
   else {
 
-    auto nextvol = ray.fVolume;
-    double snext = vecgeom::kInfLength;
-    int nsmall   = 0;
-
-    // if (generation == 2) {
-    //   printf("1-fCrtState: ");
-    //   ray.fCrtState.Print();
-    //   printf("1-fNextState: ");
-    //   ray.fNextState.Print();
-    //   ray.fVolume->Print();
-    // }
+    auto nextvol    = ray.fVolume;
+    Precision snext = vecgeom::kInfLength;
+    int nsmall      = 0;
 
     while (nextvol == ray.fVolume && nsmall < kMaxTries) {
-      snext   = BVHNavigator::ComputeStepAndPropagatedState(ray.fPos, ray.fDir, vecgeom::kInfLength, ray.fCrtState,
-                                                           ray.fNextState);
+      vecgeom::Precision dist = 0.;
+      if (rtdata.fApproach) {
+        dist = BVHNavigator::ComputeStepToApproachNextVolume(ray.fPos, ray.fDir, vecgeom::kInfLength, ray.fCrtState);
+        ray.fPos += dist * ray.fDir;
+      }
+      snext = BVHNavigator::ComputeStepAndPropagatedState(ray.fPos, ray.fDir, vecgeom::kInfLength, ray.fCrtState,
+                                                          ray.fNextState, kPush);
```

Comment:
We should not push unconditionally for every step. If it's needed for single precision, we should do it only then.

Reply:
For the geometry case, it makes sense to push unconditionally since propagation is always done always to the boundary, and before the propagation contained a push which was worse. The way it is done does not introduce biases because it just computes the distance from a pushed point, compensating for it after, and avoiding starting numerically from the wrong side of the boundary. This removes loopers also in the double-precision case in the raytracing example. In the case of the physics examples, I conditioned the push now for the single-precision case: there the danger may be only for the charged particle transport that may get wrongly pushed during the last step before crossing the boundary.